alacritty/src/config/monitor.rs

```diff
@@ -13,40 +14,45 @@ pub struct Monitor {
 }
 
 impl Monitor {
-    pub fn new<P>(path: P, event_proxy: EventProxy) -> Monitor
-    where
-        P: Into<PathBuf>,
-    {
-        let path = path.into();
-
+    pub fn new(paths: Vec<PathBuf>, event_proxy: EventProxy) -> Monitor {
         Monitor {
             _thread: thread::spawn_named("config watcher", move || {
                 let (tx, rx) = mpsc::channel();
                 // The Duration argument is a debouncing period.
                 let mut watcher =
                     watcher(tx, Duration::from_millis(10)).expect("Unable to spawn file watcher");
-                let config_path = ::std::fs::canonicalize(path).expect("canonicalize config path");
-
-                // Get directory of config.
-                let mut parent = config_path.clone();
-                parent.pop();
 
-                // Watch directory.
-                watcher
-                    .watch(&parent, RecursiveMode::NonRecursive)
-                    .expect("watch alacritty.yml dir");
+                // Get all unique parent directories.
+                let mut parents = paths
+                    .iter()
+                    .map(|path| {
+                        let mut path = fs::canonicalize(path).expect("canonicalize config path");
```

Comment:
Why would we not warn users if an imported path doesn't exist?

Reply:
I mean in message bar at least, I feel like you can have a cross platform config which sources config for each machine, but some of them are not presented depending on a machine. In general I want to simulate `/tmp/alacritty/*` which will try to source every config in that `/tmp/alacritty`, which logically won't warn if directory is empty, since it's also expected, but I'm not sure if you really want to support regex like things here, so that's why I've suggested not to warn if the user specified config is not there.

Assuming that users usually know that something in that particular path exists and put that line(even with missing path) on purpose for whatever reason, that sound logically to me, when I was suggesting it.

Reply:
That seems a bit excessive. The "normal" usecase is to override one global config, which would obviously always be present. So warning when the path is incorrect would be extremely useful.

Reply:
Ok.